.github/workflows/release.yaml

```diff
@@ -120,30 +130,34 @@ jobs:
         docker rm -f "${container}"
         ls -la artifacts/*
 
-    - name: Build Package Tester
-      uses: docker/build-push-action@v2
-      with:
-        context: .
-        load: true
-        platforms: ${{ matrix.os.name }}/${{ matrix.arch.platform }}
-        file: dist/${{ matrix.os.pkg_type }}.dockerfile
-        target: tester
-        build-args: |
-          OS_NAME=${{ matrix.os.distro }}
-          OS_VERSION=${{ matrix.os.version }}
-          PG_VERSION=${{ matrix.postgres.version }}
-          RUST_VERSION=${{ env.RUST_VERSION }}
-          RELEASE_FILE_NAME=${{ steps.metadata.outputs.filename }}
-        secrets: |
-          "AWS_ACCESS_KEY_ID=${{ secrets.PROMSCALE_EXTENSION_SCCACHE_AWS_ACCESS_KEY_ID }}"
-          "AWS_SECRET_ACCESS_KEY=${{ secrets.PROMSCALE_EXTENSION_SCCACHE_AWS_SECRET_ACCESS_KEY }}"
-        tags: ${{ steps.metadata.outputs.image }}-test
-
-    - name: Test Package
+    - name: Test package against all variants
       env:
-        EXTENSION_DOCKER_IMAGE: ${{ steps.metadata.outputs.image }}-test
+        PG_VERSION: ${{ matrix.postgres.version }}
+        DOCKER_PLATFORM: ${{ matrix.os.name }}/${{ matrix.arch.platform }}
+        PACKAGE_TO_TEST: artifacts/${{ steps.metadata.outputs.outfile }}
+        IMAGE_BASE: ${{ steps.metadata.outputs.image }}-test
+        DOCKER_FILE: dist/${{ matrix.os.pkg_type }}.test.dockerfile
         CONNECTOR_DOCKER_IMAGE: "timescale/promscale:0.11.0-alpha"
-      run: ./tools/smoke-test "${EXTENSION_DOCKER_IMAGE}" "${CONNECTOR_DOCKER_IMAGE}" "${DOCKER_PLATFORM}"
+      run: |
```

Comment:
IMO this is script is too big to keep in GH action

Reply:
Agree. I would like to punt this as a tech-debt item to rework later. I will open an issue for it.

Reply:
https://github.com/timescale/promscale_extension/issues/252